### PR TITLE
Add hostname to /etc/hosts

### PR DESCRIPTION
**Pull request by meyskens**
To prevent `unable to resolve host` when using sudo

### CODE REVIEW

**Comment by moul** [comment]
See #52 and #33 :)

I actually did the same patch earlier so I understand quite well the usefulness, before closing this PR, let's discuss if we can find an alternative solution so everybody is happy :)

**Comment by meyskens** [comment]
Could we maybe only apply it on first boot? That way it can't mess up user changes.

**Comment by moul** [comment]
This can be a solution,

Note also, `${HOSTNAME}` may vary from a boot to another if you change your server details from the console, this point doesn't matter so much I think, changing server details is already an advanced usage

@aimxhaisse, @brmzkw, @achamo, do you have an opinion ?

**Comment by moul** [comment]
After reviewing old issues and discussing with @QuentinPerez, we decided to configure /etc/hosts only on the first boot

We need to be sure that a file manually edited won't be overridden automatically on a next boot

So you need to update your script so it detects if it can update the file or not, we can maybe just look for the presence of "127.0.0.1 server", I'm not 100% sure it will handle all cases

Also, you will need to update the localhost ipv6 entry for the future ipv6 release

**Comment by meyskens** [comment]
Looking for the existing line is a good idea i guess, that will probably also handle hostname changes.

**Comment by moul** [comment]
Looks good, but it will only append new `localhost` lines,

I think it may be useful to also comment old `localhost` lines too

**Comment by meyskens** [comment]
While I was working on that I was thinking that might cause problems with people adding their own names like `127.0.0.1 database`. I have seen some tutorials do that in the past.

**Comment by moul** [comment]
I think we only need to take care about the first boot, after that, if the user wants to change its hostname, he will needs to update its files by himself (as we often do on standard dedicated servers)

Maybe you can just check for `127.0.0.1 server` which is the default

**Comment by meyskens** [comment]
You mean the exact word `server`?

**Comment by moul** [comment]
yes :)

**Comment by meyskens** [comment]
I tried to keep the if line to even add the names in a rare case where `127.0.0.1 server` wasn't present but realized it won't work the way i kept it there. Want me to remove it or try to keep it and make it check for non "localhost" 127.0.0.1 lines?

**Comment by moul** [comment]
Your actual solution looks good to me, we need to test it :)

**Comment by moul** [comment]
@QuentinPerez can you test it on your dev bootscript ?

File: Linux/tree-common/init [hunk 1]
```diff
@@ -249,6 +249,12 @@ log_begin_msg "Creating default configuration files on rootfs..."
 run mkdir -p "${rootmnt}/etc"
 run touch ${rootmnt}/etc/hostname ${rootmnt}/etc/resolv.conf
 [ -s ${rootmnt}/etc/hostname ] || run sh -ec "echo ${HOSTNAME} > ${rootmnt}/etc/hostname"
+if ! $(cat /etc/hosts | grep -q "127.0.0.1"); then
+echo '127.0.0.1 '${HOSTNAME} >>/etc/hosts
+echo '::1 '${HOSTNAME} >>/etc/hosts
+fi
+sed -i "s/127.0.0.1.*server/127.0.0.1 ${HOSTNAME}/g" /etc/hosts
+sed -i "s/::1.*server/::1 ${HOSTNAME}/g" /etc/hosts
```
Comment (QuentinPerez) [inline Linux/tree-common/init]: @meyskens can you prefix the paths with `${rootmnt}`
e.g. : `${rootmnt}/etc/hosts`

File: Linux/tree-common/init [hunk 2]
```diff
@@ -249,6 +249,12 @@ log_begin_msg "Creating default configuration files on rootfs..."
 run mkdir -p "${rootmnt}/etc"
 run touch ${rootmnt}/etc/hostname ${rootmnt}/etc/resolv.conf
 [ -s ${rootmnt}/etc/hostname ] || run sh -ec "echo ${HOSTNAME} > ${rootmnt}/etc/hostname"
+if ! $(cat /etc/hosts | grep -q "127.0.0.1"); then
```
Comment (QuentinPerez) [inline Linux/tree-common/init]: Can you add a test to handle the case where we found the address `127.0.1.1` to set the hostname at this line

https://www.debian.org/doc/manuals/debian-reference/ch05.en.html#_the_hostname_resolution

**Comment by QuentinPerez** [comment]
LGTM

**Comment by QuentinPerez** [comment]
@meyskens with your patches, when I reboot my server the information are appended

```
# after one reboot
$> cat /etc/hosts
127.0.1.1 scw-d75c55
127.0.0.1       localhost server
::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters127.0.1.1 scw-d75c55
::1 scw-d75c55
127.0.1.1 scw-d75c55
::1 scw-d75c55
```

Can you find a way to fix that ? 😊

**Comment by moul** [comment]
`$(cat /etc/hosts | grep -q "127.0.1.1"); then` should be `$(cat ${rootmnt}/etc/hosts | grep -q "127.0.1.1"); then`

**Comment by moul** [comment]
Hi @meyskens,

Thanks for your work on this subject,
but we used a different way to address this issue,
sorry for not having merge your work 🍺

**Pull request closed without merging**